catroid/src/main/java/org/catrobat/catroid/devices/arduino/ArduinoImpl.java

```diff
@@ -187,6 +187,10 @@ private void tryInitialize() throws IOException, SerialException {
 			sendFirmataMessage(new SetPinModeMessage(pin, SetPinModeMessage.PIN_MODE.PWM.getMode()));
 		}
 		reportSensorData(true);
+
+		// get status of digital ports
+		sendFirmataMessage(new ReportDigitalPortMessage(0, true));
+		sendFirmataMessage(new ReportDigitalPortMessage(1, true));
```

Comment:
After some testing I removed it in my most recent commit, as it is unnecessary.
I initially thought this was necessary to get the current state of the outputs (e.g. after restarting a program) - but this actually works without these two lines. I'm not sure why, but I suppose this is because AsuroImpl keeps running in the background even if  the user's program is restarted?